payara/Payara
# FISH-6966: Package Opentelemetry into distribution

## Description

This brings OpenTelemetry APIs and default implementations into server as basis for implementing MP Telemetry and redoing opentracing support via shim.

Applications can now manually bootstrap OTEL SDK and use the tracer. OTLP export via GRPC and HTTP works. It is possible to add additional components into `domaindir/lib` and have autoconfiguration bootstrap them properly.

## Testing
### New tests
New samples module will be basis for adding arquillian tests. So far the test has been only manual using this exact application.

### Testing Performed

* Deploy the sample application, observe OTLP traces appearing in Jaeger (via its native OTLP collection endpoint)
* Configure tracing via API
* Configure tracing via autoconfiguration, providing properties from MP config (one of MP Tracing requriements)
* Add jar for logging exporter into domaindir/lib, configure it via microprofile config.
